src/test/java/BaseTest.java

```diff
@@ -1,11 +1,125 @@
+        options.addArguments("--remote-allow-origins=*");
+
+        //Manage Browser - wait for 10 seconds before failing/quitting
+        driver = new ChromeDriver(options);
+        //driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+
+        actions = new Actions(driver);
+
+        //Navigate to Url
+        url = BaseUrl;
+        navigatorUrl();
+    }
+
+    @AfterMethod
+    public void closeBrowser(){
+        driver.quit();
+    }
+
+
+    void provideEmail(String email){
+        //WebElement emailField = driver.findElement(By.cssSelector("input[type='email']"));
+        WebElement emailField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='email']")));
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+    //"demo@class.com"
+
+    void providePassword(String password){
+        //WebElement passwordField = driver.findElement(By.cssSelector("input[type='password']"));
+        WebElement passwordField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='password']")));
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+    //"te$t$tudent"
+
+    void clickSubmit(){
+        //WebElement submitButton = driver.findElement(By.cssSelector("button[type='submit']"));
+        WebElement submitButton = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("button[type='submit']")));
+        submitButton.click();
+    }
+
+    //Helper Method
+    public void navigatorUrl(){
+        driver.get(url);
+    }
+
+    /*
+    *Test Script using Page Object Model.
+    *Locators used in these test were located using By abstract class.
+    */
+
+
+    @Test
+    public void loginWithCorrectCredentials(){
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+
+        loginPage.login();
+//        loginPage.provideEmail("demo@class.com");
+//        loginPage.providePassword("te$t$tudent");
+//        loginPage.clickSubmit();
+
+        Assert.assertTrue(homePage.getUserAvatarIcon().isDisplayed());
+    }
+    @Test
+    public void loginWithIncorrectCredentials() throws InterruptedException{
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+
+
+        loginPage.provideEmail("demo1@class.com");
+        loginPage.providePassword("te$t$tudent1");
+        loginPage.clickSubmit();
+
+        //Assert.assertTrue(homePage.getUserAvatarIcon().isDisplayed());
+        Assert.assertEquals(driver.getCurrentUrl(),url);
+//        try {
+//            Assert.assertTrue(homePage.getUserAvatarIcon().isDisplayed());
+//        }catch(NoSuchElementException e){
+//            System.out.println("Element not displayed as expected");
+//        }
+    }
```

Comment:
**HINT**

1. **BasePage and BaseTest** 

**BaseTest** — Implements the necessary methods for work with webdriver.

The BaseTest class contains common setup and teardown methods, as well as any shared functionality or configurations required by multiple test classes.

**BaseTest**

The problem we are trying to solve here is the reuse of common behaviors across test classes avoiding coding duplication and centralizing these actions at one point.

**@BeforeSuite**	- The annotated method will be run before all tests in this suite have run.

**@BeforeMethod** -	The annotated method will be run before all the test methods in the current class have been run.

**@AfterMethod**	- The annotated method will be run after each test method.

_Example without a Base Test Class_

![image](https://user-images.githubusercontent.com/104761482/233972627-2d7ffdc7-6a2f-4e9f-8a3a-a1b4a7a47bc4.png)


**BasePage**  — class serves as a foundation or parent class for individual page classes. It contains common methods and functionalities that are shared across multiple pages in an application.

_Example without a Base Page Class_

![image](https://user-images.githubusercontent.com/104761482/236891493-7f7013c7-fde2-4bf9-9084-3aa7483df8e3.png)


2. **Test in Page Object Model (POM)**

![Clean_Test](https://user-images.githubusercontent.com/104761482/236863813-f30d7bfc-e89d-4419-bdea-4b1517155e39.png)